src/pages/GovPages/Proposal/index.tsx

```diff
@@ -0,0 +1,282 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { ThemeContext } from 'styled-components'
+import { useTranslation } from 'react-i18next'
+import { Flex } from 'rebass'
+import { ETHER } from 'dxswap-sdk'
+import { ChevronLeft } from 'react-feather'
+
+import { USDC } from '../../../constants'
+import { TYPE } from '../../../theme'
+import { PageWrapper } from '../../Pool/styleds'
+import { RowBetween } from '../../../components/Row'
+import { AutoColumn } from '../../../components/Column'
+import CurrencyLogo from '../../../components/CurrencyLogo'
+import HourGlass from '../../../assets/svg/hourglass.svg'
+import Done from '../../../assets/svg/done_all.svg'
+import Block from '../../../assets/svg/block.svg'
+import {
+  Container,
+  ContentTitle,
+  InfoText,
+  TextCard,
+  ContentCard,
+  ProgressDiv,
+  VoteButton,
+  ProposerAddress
+} from './styleds'
+import { useRouter } from '../../../hooks/useRouter'
+import { fakeProposalData } from '../constant'
+import { Failed, InProgress, Passed, VoteStatus } from '../VoteStatus'
+
+const pairName = 'USDC/ETH'
+const currency = USDC
+const currency1 = ETHER
+
+export default function GovernanceProposals() {
+  const router = useRouter()
+  const { t } = useTranslation()
+  const theme = useContext(ThemeContext)
+  const proposalInfo = fakeProposalData[parseInt(router.query.id) - 1]
+  const isEnded = Date.now() > proposalInfo.until
+  const voteStatus = !isEnded ? InProgress : proposalInfo.for > proposalInfo.against ? Passed : Failed
+
+  const [isVoted, setIsVoted] = useState<{ isVoted: boolean; result: null | boolean }>({
+    isVoted: false,
+    result: null
+  })
+  const [totalVote, setTotalVote] = useState({ for: proposalInfo.for, against: proposalInfo.against })
+  const [percentage, setPercentage] = useState({ for: '0%', against: '0%' })
+
+  useEffect(() => {
+    setPercentage({
+      for: ((totalVote.for / (totalVote.for + totalVote.against)) * 100).toFixed(0) + '%',
+      against: ((totalVote.against / (totalVote.for + totalVote.against)) * 100).toFixed(0) + '%'
+    })
+  }, [totalVote])
+
+  const onBack = () => router.history.goBack()
```

Comment:
`useCallback` will for sure improve performance, since it won't trigger a rerender of any child components that use the callback every time this component rerenders. Would use it below too.